python/tank/path_cache.py

```diff
@@ -644,9 +662,17 @@ def _replay_folder_entities(self, cursor, log, max_event_log_id, ids=None):
                                    SG_ENTITY_TYPE_FIELD, 
                                    SG_ENTITY_NAME_FIELD],
                                   [{"field_name": "id", "direction": "asc"},])
+        elif ids == []:
+            # incremental sync but with no folders
+            self._log_debug(log, "No folders need to be replayed, won't fetch anything from Shotgun...")
+            sg_data = []
```

Comment:
can I suggest defining sg_data outside the if-else block - it would be more robust should the code change at a later date.